genai-perf/genai_perf/profile_data_parser/llm_profile_data_parser.py

```diff
@@ -330,37 +339,54 @@ def _preprocess_response(
-                        if isinstance(data, list) and len(data) > 0:
-                            data[0]["generated_text"] = merged_text  # type: ignore
-                    else:
-                        merged_text = "".join(
-                            [self._extract_text_output(r) for r in responses]
-                        )
-                        if self._response_format == ResponseFormat.OPENAI_COMPLETIONS:
-                            data["choices"][0]["text"] = merged_text
+                        res_outputs[i]["response"] = ""
+                        parser_result.failed += 1
+
+                try:
+                    if len(responses) > 1:
+                        data = load_json_str(remove_sse_prefix(responses[0]))
+                        if self._response_format == ResponseFormat.TRITON_GENERATE:
+                            merged_text = "".join(
+                                [
+                                    self._extract_generate_text_output(r)
+                                    for r in responses
+                                ]
+                            )
+                            data["text_output"] = merged_text
+                        elif (
+                            self._response_format == ResponseFormat.HUGGINGFACE_GENERATE
+                        ):
+                            merged_text = "".join(
+                                [
+                                    self._extract_huggingface_generate_text_output(r)
+                                    for r in responses
+                                ]
+                            )
+                            if isinstance(data, list) and len(data) > 0:
+                                data[0]["generated_text"] = merged_text  # type: ignore
                         else:
-                            data["choices"][0]["delta"]["content"] = merged_text
-                    res_outputs[i] = {"response": orjson.dumps(data).decode("utf-8")}
-                elif self._is_empty_response(responses[0]):
+                            merged_text = "".join(
+                                [self._extract_text_output(r) for r in responses]
+                            )
+                            if (
+                                self._response_format
+                                == ResponseFormat.OPENAI_COMPLETIONS
+                            ):
+                                data["choices"][0]["text"] = merged_text
+                            else:
+                                data["choices"][0]["delta"]["content"] = merged_text
+                        res_outputs[i] = {
+                            "response": orjson.dumps(data).decode("utf-8")
+                        }
+                    elif self._is_empty_response(responses[0]):
+                        res_outputs[i]["response"] = ""
+                except Exception as e:
+                    logger.debug(f"Error parsing a response: {str(e)}")
```

Comment:
Can you print any more information about the response here? Should this be logger.error? Also, will the response then be excluded from the metrics?